bot/exts/info/code_snippets.py

```diff
@@ -0,0 +1,237 @@
+import logging
+import re
+import textwrap
+from urllib.parse import quote_plus
+
+from discord import Message
+from discord.ext.commands import Cog
+
+from bot.bot import Bot
+from bot.utils.messages import wait_for_deletion
+
+log = logging.getLogger(__name__)
+
+GITHUB_RE = re.compile(
+    r'https://github\.com/(?P<repo>.+?)/blob/(?P<path>.+/.+)'
+    r'#L(?P<start_line>\d+)([-~]L(?P<end_line>\d+))?\b'
+)
+
+GITHUB_GIST_RE = re.compile(
+    r'https://gist\.github\.com/([^/]+)/(?P<gist_id>[^\W_]+)/*'
+    r'(?P<revision>[^\W_]*)/*#file-(?P<file_path>.+?)'
+    r'-L(?P<start_line>\d+)([-~]L(?P<end_line>\d+))?\b'
+)
+
+GITLAB_RE = re.compile(
+    r'https://gitlab\.com/(?P<repo>.+?)/\-/blob/(?P<path>.+/.+)'
+    r'#L(?P<start_line>\d+)([-](?P<end_line>\d+))?\b'
+)
+
+BITBUCKET_RE = re.compile(
+    r'https://bitbucket\.org/(?P<repo>.+?)/src/(?P<ref>.+?)/'
+    r'(?P<file_path>.+?)#lines-(?P<start_line>\d+)(:(?P<end_line>\d+))?\b'
+)
+
+
+class CodeSnippets(Cog):
+    """
+    Cog that parses and sends code snippets to Discord.
+
+    Matches each message against a regex and prints the contents of all matched snippets.
+    """
+
+    async def _fetch_response(self, url: str, response_format: str, **kwargs) -> str:
+        """Makes http requests using aiohttp."""
+        try:
+            async with self.bot.http_session.get(url, **kwargs) as response:
+                if response_format == 'text':
+                    return await response.text()
+                elif response_format == 'json':
+                    return await response.json()
+        except Exception:
+            log.exception(f'Failed to fetch code snippet from {url}.')
```

Comment:
Use `log.error` instead.

Reply:
The log should also contain the HTTP status code from the response (which you can get by catching the correct error as mentioned in the other comment).

Reply:
Is what I did in https://github.com/python-discord/bot/pull/1028/commits/08b793024f271de009aab2391cd85576af5313cf fine? I feel like the error message may get a bit long, so should I split it up into two lines instead?